source/2_platform/rex_opengl/src/resources/shader_program.cpp

```diff
@@ -110,14 +107,34 @@ namespace rex
     GL_CALL(glDeleteShader(fragment_shader_id));
   }
 
+  //-------------------------------------------------------------------------
+  ShaderProgram::ShaderProgram(ShaderProgram&& other) noexcept
+      : m_resource_id(rsl::exchange(other.m_resource_id, 0))
+      , m_uniform_locations(rsl::exchange(other.m_uniform_locations, rsl::unordered_map<rsl::small_stack_string, ShaderUniformLocation>()))
```

Comment:
this can just be a move, doesn't have to be an exchange

Reply:
Yes but to keep it consistent with the other lines I would like to keep rsl::exchange